File: CMDB/CMDB.UI.Specflow/Questions/Main/OpenTheAccountTypeOverviewPage.cs
using Bright.ScreenPlay.Actors;
using CMDB.UI.Specflow.Abilities.Pages;
using Bright.ScreenPlay.Questions;
using CMDB.UI.Specflow.Abilities.Pages.Types;

namespace CMDB.UI.Specflow.Questions.Main
{
    public class OpenTheAccountTypeOverviewPage : Question<TypeOverviewPage>
    {
        public override TypeOverviewPage PerformAs(IPerformer actor)
        {
            var page = actor.GetAbility<MainPage>();
            page.WaitUntilElmentVisableByXpath("//a[@id='Types']");
            page.ClickElementByXpath("//a[@id='Types']");
            page.ClickElementByXpath("//a[@id='Account Type34']");
            page.ClickElementByXpath("//a[@href='/AccountType']");
            page.WaitOnAddNew();
            return new();
        }
    }
}

File: Java/playwright/src/test/java/be/hans/cmdb/Questions/AccountType/OpenTheAccountTypeOverviewPage.java
package be.hans.cmdb.Questions.AccountType;

import be.brightest.ScreenPlay.Abilities.WebPageFactory;
import be.brightest.ScreenPlay.Actors.IActor;
import be.brightest.ScreenPlay.Question.Question;
import be.cmdb.pages.MainPage;
import be.cmdb.pages.Type.TypeOverviewPage;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.AriaRole;

public class OpenTheAccountTypeOverviewPage extends Question<TypeOverviewPage> {
    @Override
    public TypeOverviewPage performAs(IActor actor) {
        MainPage ability = (MainPage) actor.getAbility(MainPage.class);
        ability.getPage().getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Types")).click();
        ability.getPage().locator("xpath=//a[@id='Account Type34']").click();
        ability.getPage().getByRole(AriaRole.LINK, new Page.GetByRoleOptions().setName("Overview")).click();
        ability.newButton().waitFor();
        return WebPageFactory.create(TypeOverviewPage.class, ability.getPage());
    }
}
